### DISCUSSION TITLE
How does pylance decide what is and isn't exported from a module?

### DISCUSSION THREAD

**Discussion by decorator-factory** [microsoft/pylance-release#3397] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/42166884/192380388-b4c9cca5-a8c0-4ce6-acb6-651d7a018fea.png)

Source of `textual/views/__init__.py`:
```py
from ._dock_view import DockView, Dock, DockEdge
from ._grid_view import GridView
from ._window_view import WindowView
```
I think this is quite a common way of re-exporting stuff. Is it really required to also add an `__all__`?

**Comment by bschnurr** [comment, marked as answer]
see
https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface

Can use `__all__` or redundant import `from ._dock_view import DockView as DockView`

**Reply by erictraut** [reply]
The rules differ depending on whether you're talking about a stub (".pyi") file, a ".py" file within a "py.typed" library, or a normal ".py" file (that's not part of a "py.typed" library). For stub files, the behavior is dictated by PEP 484. For ".py" files in a "py.typed" library, stub-like rules apply as documented in the documentation that @bschnurr referenced above. For regular ".py" files, all imported symbols are assumed to be re-exported unless they are explicitly named using "private" naming conventions. In your screen shot above, I'm assuming that the `textual.view` submodule is part of a "py.typed" library.

**Reply by decorator-factory** [reply]
I see, thank you. So is this a problem with `textual`? Should I make an issue to make re-exports more explicit to linters?
